enhancements/installer/packet-ipi.md

```diff
@@ -0,0 +1,197 @@
+running Red Hat OpenShift 4 on a horizontally scalable bare metal architecture
+in data centers.
+
+As a first step, we would add the Packet IPI to the installer codebase. With accompanying documentation, this makes a the Packet option available through the CLI. This would enable users to deploy OpenShift with IPI in a way
+that is very similar to UPI but simplifies the process.
+
+Following other IPI installers, this first step would include
+
+- Making Packet IPI documentation available here:
+  <https://github.com/openshift/installer/blob/master/docs/user/packet/install_ipi.md>
+- Adapt Packet's sig-lifecyclce ClusterAPI v1alpha2 for use as an OpenShift's ClusterAPI v1beta1 Machine driver
+- Prepare the Terraform code and Packet types necessary for an IPI installer
+- Making a CI job executing the provisioning scripts to test the IPI installer
+
+### Non-Goals
+
+It is outside the scope of this enhancement to provide explanations about the
+installation of infrastructure elements that are considered as required and
+owned by the user (e.g. DNS, Load Balancer).
+
+Additional enhancement requests will be created to determine the best approach for fulfilling each need.
+
+## Proposal
+
+- Define and implement Packet types in the openshift/installer
+  - High level variables should include:
+    - API Key
+    - Project ID
+  - Boostrap node variables include:
+    - Device plan (defaulting to minimum required)
+    - Facility
+    - Virtual Networks
+  - Control plane variables rely on boostrap variables and add:
+    - Device plan (defaulting to provide a good experience)
+  - Additional clusters creation should include
+    - Device plan (defaulting to provide a good experience)
+    - Facility
+    - Virtual Networks
+    - Future: IP Reservations, Hardware Reservations, Spot Market Requests
+- Write documentation to help users use and understand the Packet installer, to include:
+  - Packet usage basics (accounts and API keys)
+  - Packet resource requirements and options
+  - OpenShift Installer options
+  - Non-Packet components that build on the experience (DNS configuration)
+- Setup the CI job to have a running test suite
+
+### Implementation Details/Notes/Constraints
+
+- RHCOS must be made available and maintained.
+  Images available for provisioning may be custom or official.
+  Official images can be hosted outside of Packet, such as ContainerLinux and RancherOS. More details area available here: <https://github.com/packethost/packet-images>
+  Custom Images can be hosted on git, as described here:
+  - <https://www.packet.com/developers/docs/servers/operating-systems/custom-images/>
+  - <https://www.packet.com/developers/changelog/custom-image-handling/>
+- Packet does not offer a managed DNS solution. Route53, CloudFlare, or RFC 2136
+  (DNS Update), and NS record based zone forwarding are considerations. Existing
+  IPI drivers like BareMetal have similar limitations.
```

Comment:
For `baremetal` IPI, we do some automation of required DNS records for use within the cluster itself.  I think that should be reusable here if multicast is allowed on packet's network.  Has this been considered?

Reply:
Multicast is not fully supported, it may work enough for our needs.  Are you suggesting mDNS?

Reply:
Yeah.  Take a look at the "internal DNS" section of https://github.com/openshift/enhancements/blob/master/enhancements/network/baremetal-networking.md

Reply:
mdns is used internally for IPI. Both for nodes resolution and for pointing to the api-int entry.
It has nothing to do with external consumption.

In assisted-installer (where we also rely on BM IPI ignitions) we do provide optional route53 based dns to support external api + *.apps resolution.

Reply:
Interesting. I've been looking for something like https://github.com/openshift/mdns-publisher

(for https://github.com/kubernetes-sigs/external-dns/issues/1604)

Reply:
https://github.com/openshift/machine-config-operator/blob/master/templates/common/baremetal/files/baremetal-mdns-publisher.yaml

This is the relevant repo that provides networking related manifests as part of IPI BareMetal platfrom - you'll find there mdns, keepalived and coredns.